lms/djangoapps/instructor/tests/test_enrollment.py

```diff
@@ -494,6 +497,95 @@ def test_delete_submission_scores_attempts_children(self):
+        cls.chapter = ItemFactory.create(
+            parent=cls.course,
+            category="chapter",
+            display_name="Test Chapter"
+        )
+        cls.sequence = ItemFactory.create(
+            parent=cls.chapter,
+            category='sequential',
+            display_name="Test Sequential 1",
+            graded=True
+        )
+        cls.vertical = ItemFactory.create(
+            parent=cls.sequence,
+            category='vertical',
+            display_name='Test Vertical 1'
+        )
+        cls.problem = ItemFactory.create(
+            parent=cls.vertical,
+            category="problem",
+            display_name="Test Problem"
+        )
+        cls.user = UserFactory()
+
+    def test_delete_student_state(self):
+        problem_location = self.problem.location
+        # Create a student module for the user
+        StudentModule.objects.create(
+            student=self.user,
+            course_id=self.course_key,
+            module_state_key=problem_location,
+            state=json.dumps({})
+        )
+
+        # Create a submission and score for the student using the submissions API
+        student_item = {
+            'student_id': anonymous_id_for_user(self.user, self.course_key),
+            'course_id': self.course_key.to_deprecated_string(),
+            'item_id': problem_location.to_deprecated_string(),
+            'item_type': 'openassessment'
+        }
+        submission = sub_api.create_submission(student_item, 'test answer')
+        sub_api.set_score(submission['uuid'], 1, 2)
+        subsection_grade_factory = SubsectionGradeFactory(
+            self.user,
+            self.course,
+            get_course_blocks(self.user, self.course.location)
+        )
+        grade = subsection_grade_factory.create(self.sequence)
+        self.assertEqual(grade.all_total.earned, 1)
+        self.assertEqual(grade.graded_total.earned, 1)
+        self.assertEqual(grade.all_total.possible, 2)
+        self.assertEqual(grade.graded_total.possible, 2)
+
+        # Delete student state using the instructor dash
+        reset_student_attempts(
+            self.course_key,
+            self.user,
+            problem_location,
+            requesting_user=self.user,
+            delete_module=True,
+        )
+
+        # Verify that the student's scores have been reset in the submissions API
+        score = sub_api.get_score(student_item)
+        self.assertIs(score, None)
+        # Verify that the student's grades are reset
+        subsection_grade_factory = SubsectionGradeFactory(
+            self.user,
+            self.course,
+            get_course_blocks(self.user, self.course.location)
+        )
+        grade = subsection_grade_factory.create(self.sequence)
+        self.assertEqual(grade.all_total.earned, 0)
+        self.assertEqual(grade.graded_total.earned, 0)
+        self.assertEqual(grade.all_total.possible, 0)
+        self.assertEqual(grade.graded_total.possible, 0)
```

Comment:
I thought 'possible' reflected the total number of points it would be possible to earn across the questions _answered_ within a subsection. Since none of the questions are answered after the student state is deleted, `possible` returns to 0.

To verify this I will retrieve the initial grade for the subsection before any answer is submitted, as well, and create a helper to DRY-ify this code.